### PR TITLE
Merge localization side branch to Revit 2014

### PR DESCRIPTION
**Pull request by Benglin**
This is the equivalent of bringing `Dynamo:localization` into `Dynamo:master`, it is meant to bring `DynamoRevit:localization2014` into `DynamoRevit:Revit2014`.

This pull request must be merged almost immediately after [the main one goes into `Dynamo:master`](https://github.com/DynamoDS/Dynamo/pull/3617).

### CODE REVIEW

File: src/DynamoRevit/DynamoRevitApp.cs [hunk 1]
```diff
@@ -52,7 +52,7 @@ public Result OnStartup(UIControlledApplication application)
                 ElementBinder.IsEnabled = true;
 
                 //TAF load english_us TODO add a way to localize
```
Comment (lukechurch) [inline src/DynamoRevit/DynamoRevitApp.cs]: Remove comment? It's now obsolete.

File: src/Libraries/RevitNodes/Properties/Resources.resx [hunk 2]
```diff
@@ -117,10 +117,13 @@
   <resheader name="writer">
     <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
   </resheader>
-  <data name="App_Description" xml:space="preserve">
```
Comment (lukechurch) [inline src/Libraries/RevitNodes/Properties/Resources.resx]: Is this text now no longer used?

**Comment by lukechurch** [comment]
lgtm after comments addressed

File: src/Libraries/RevitNodes/Properties/Resources.resx (hunk 2 above)
Comment (Benglin) [inline src/Libraries/RevitNodes/Properties/Resources.resx]: Yah, the use of `App_Description` has been moved from `Libraries/RevitNodes` to `DynamoRevit`, so a new entry is being added there.

**Comment by Benglin** [comment]
Comments are addressed. Thanks a lot for your help, @lukechurch! Appreciate it.

**Pull request merged**